Zend/zend_object_handlers.c

```diff
@@ -535,9 +535,8 @@ ZEND_API uint32_t *zend_get_property_guard(zend_object *zobj, zend_string *membe
 	if (EXPECTED(Z_TYPE_P(zv) == IS_STRING)) {
 		zend_string *str = Z_STR_P(zv);
 		if (EXPECTED(str == member) ||
-		     /* "str" always has a pre-calculated hash value here */
-		    (EXPECTED(ZSTR_H(str) == zend_string_hash_val(member)) &&
```

Comment:
How about using `zend_string_hash_val` for `str` instead? Is the function is called multiple times this should be faster as it avoids the content comparison.

Reply:
I see, Nikita prefers dropping the hash comparison. I'm not gonna question him on this one :slightly_smiling_face: